qemu/tests/numa_prealloc_threads.py

```diff
@@ -0,0 +1,63 @@
+import re
+
+from virttest import error_context
+from virttest.qemu_monitor import QMPCmdError
+from avocado.utils import process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    numa_prealloc_threads test
+    1) Boot a guest with thread-context and cpu-affinity option
+    2) Obtain the thread-id
+    3) Check the affinity obtained from QEMU is correct
+    4) With sandbox enabled, try to change the cpu-affinity
+       and handle the error
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    test.log.info("Check host CPUs number")
+    host_cpus = int(process.system_output("nproc"))
+    smp_fixed = params.get_numeric("smp_fixed")
+    if host_cpus < smp_fixed:
+        test.cancel("The host only has %d CPUs, it needs at least %d!" % (host_cpus, smp_fixed))
```

Comment:
This is incorrect logic.  By default,  the framework create vm (calculate the cpu topology and launch qemu process) during env process.  


This checking step need be handled before create vm. In short,  `not_preprocess = yes` need be added in the cfg.
It can be enabled again after the checking and call `env_process.preprocess_vm`

Reply:
@yanan-fu 
Yes, you are right... updated now, thanks!